api/src/main/proto/stellarstation/api/v1/radio/radio.proto

```diff
@@ -51,11 +51,137 @@ message CommunicationProtocol {
+// Code rate (r): 1/2
+// Constraint length (K): 7
+// Connection vectors:
+//   G1 = 1111001
+//   G2 = 1011011
+message CCSDSConvolutionalCodingParams {
+  // Polynomial inversion
+  enum PolynomialInversion {
+    // CCSDS with normal phase rotation
+    G1_G2 = 0;
+
+    // CCSDS with inverted phase rotation
+    G1_G2_INVERTED = 1;
+
+    G2_INVERTED_G1 = 2;
+
+    G2_G1 = 3;
+  }
+
+  PolynomialInversion polynomial_inversion = 1;
+
+  // Enum describing different rates for puncturing
+  enum Puncturing {
+    // Puncturing unused
+    OFF = 0;
+
+    // Puncturing rate 2/3
+    PUNCTURE_2_3 = 1;
+
+    // Puncturing rate 3/4
+    PUNCTURE_3_4 = 2;
+
+    // Puncturing rate 4/5
+    PUNCTURE_4_5 = 3;
+
+    // Puncturing rate 5/6
+    PUNCTURE_5_6 = 4;
+
+    // Puncturing rate 6/7
+    PUNCTURE_6_7 = 5;
+
+    // Puncturing rate 7/8
+    PUNCTURE_7_8 = 6;
+  }
+
+  // Optional puncturing for convolutional code
+  Puncturing puncturing = 2;
+}
+
+// CCSDS-specific scrambling parameters.
+// Additive scrambler with pseudo-random sequence generated by the polynomial:
+// h(x) = x^8 + x^7 + x^5 + x^3 + 1
+message CCSDSScramblingParams {
+}
+
+// CCSDS-specific Reed-Solomon parameters
+message CCSDSReedSolomonParams {
+  // BlockLengthType
+  enum BlockLengthType {
+    // (255, 223, 16) RS code
+    RS_223_255 = 0;
+
+    // (255, 239, 8) RS code
+    RS_239_255 = 1;
+  }
+
+  BlockLengthType block_length_type = 1;
+}
+
+// https://public.ccsds.org/Pubs/131x0b3e1.pdf
+//
+// CCSDS transfer frames allow for a few common variations satellites use such as setting a
+// custom synchronization word, inverting the convolutional coding polynomials,
+// etc. The sequence of error-correcting codes will be applied as they are
+// defined in the CCSDS standard.
+message CCSDSTransferFrame {
+  // Synchronization word parameters. If unset, synchronization will be enabled
+  // using the default CCSDS transfer frame standard synchronization word as in
+  // 9.3.5 (Page 9-3) in the specification.
+  SynchronizationParams synchronization_params = 1;
+
+  // Convolutional coding parameters. Disabled if unset.
+  CCSDSConvolutionalCodingParams convolutional_coding_params = 2;
+
+  // Scrambling parameters. Disabled if unset.
+  CCSDSScramblingParams scrambling_params = 3;
+
+  // Reed-Solomon parameters. Disabled if unset.
+  CCSDSReedSolomonParams reed_solomon_params = 4;
+
+  // Frame length in bytes
+  uint32 frame_length_bytes = 5;
+
+  // Line Coding used.
+  LineCoding line_coding = 6;
+}
+
+// Bitstream for satellites that don't conform to a particular
+// standard. Usually, only a synchronization word and expected
+// frame length can be set.
+message Bitstream {
+  // Synchronization word parameters. Disabled if unset.
+  SynchronizationParams synchronization_params = 1;
+
+  // Frame length in bytes
+  uint32 frame_length_bytes = 2;
+
+  // Line Coding used.
+  LineCoding line_coding = 3;
+}
+
+// SynchronizationParams contains parameters for frame
```

Comment:
We don't use golang style, so just something like `Parameters for frame synchronization.`